src/main/java/sally/task/Task.java

```diff
@@ -0,0 +1,67 @@
+package sally.task;
+
+import sally.exception.SallyException;
+import sally.storage.Storage;
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
+    protected Type taskType;
+    protected String moreInfo;
+    protected boolean saveTask;
+    protected Storage storage;
+
+    protected enum Type {
+        TODO, DEADLINE, EVENT
+    }
+
+    //Todo
+    public Task(String description, boolean saveTask) {
+        this.description = description;
+        this.saveTask = saveTask;
+    }
+
+    //sally.task.Event and sally.task.Deadline
+    public Task(String description, String moreInfo, boolean saveTask) {
+        this.description = description;
+        this.moreInfo = moreInfo;
+        this.saveTask = saveTask;
+    }
+
+    public String getMoreInfo() {
+        return this.moreInfo;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " ");
+    }
+
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    public void markAsUndone() {
+        this.isDone = false;
+    }
+
+    public boolean getDoneStatus() {
```

Comment:
As much as possible, it would be nice to name booleans to sound like booleans. Perhaps you could name this `isDone` instead? :)